game/player/player.py

```diff
@@ -0,0 +1,102 @@
+from abc import abstractmethod
+from dataclasses import dataclass
+from threading import Thread, Semaphore
+
+from client.game_client import GameClient
+from entity.tanks.tank import Tank
+from map.map import Map
+
+
+@dataclass
+class Player(Thread):
+    __type_order = ('spg', 'light_tank', 'heavy_tank', 'medium_tank', 'at_spg')
+    __possible_colours = ('blue', 'black', 'green')
+
+    def __init__(self, name: str, password: str, is_observer: bool,
+                 turn_played_sem: Semaphore, current_player: list[1], player_index: int):
+        super().__init__(daemon=True)
+        self.idx: int = -1
+        self.name = name
+        self.password = password
+        self.is_observer: bool = is_observer
+        self.next_turn_sem = Semaphore(0)
```

Comment:
what is the point of this semaphore?

Reply:
To ensure player synchronization with the updated state of a new game turn, notified by the game logic processing thread. An alternative would be polling, but this is more CPU intensive.